src/data/equiv/algebra.lean

```diff
@@ -328,3 +327,75 @@ instance symm.is_ring_hom {e : α ≃r β} : is_ring_hom e.to_equiv.symm := hom
   e.symm.to_equiv x = e.to_equiv.symm x := rfl
 
 end ring_equiv
+
+-- We first define the type of automorphisms on groups, additive monoids,
+-- additive groups and rings using `mul_equiv`, `add_equiv` and `ring_equiv`.
+-- In each case this type is also a group.
+
+namespace group_aut
+
+def aut (γ : Type) [group γ] := mul_equiv γ γ
+
+@[extensionality] lemma mul_equiv.ext {α β : Type} [group α] [group β]
+{f g : mul_equiv α β} (h : f.to_fun = g.to_fun) : f = g :=
+by { cases f, cases g, congr, apply equiv.eq_of_to_fun_eq h }
+
+-- The group operation is defined by λ g h, mul_equiv.trans h g so that
```

Comment:
Maybe make this a doc-comment, using `/-- ... -/` so it stays attached to the definition and can appear in tool-tips, etc. Doc-comments should make sense without much context, so probably should begin with "The group operation on automorphisms of a group is defined by ...".

Reply:
done